### PR TITLE
Rename package from bom-squad/protobom to protobom/protobom

### PR DESCRIPTION
**Pull request by eddiezane**
This repo was migrated from https://github.com/bom-squad/protobom.

This PR updates references of `bom-squad` and `bomsquad` to `protobom` acorss the repo.

This will require coordination with https://github.com/bomctl/bomctl and any other downstream consumers.

### CODE REVIEW

File: pkg/sbom/node_test.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func TestNodeFlatString(t *testing.T) {
...
+			"extref:(t)56(u)http://github.com/external(c)GitHub Link:identifiers[1]:pkg:/apk/wolfi/bash@4.1.11:originator:n(The Open Source Authors)o(false)email(noone@example.com)url(http://github.com/example)p(987-5432):protobom.protobom.Node.build_date:1700166898:protobom.protobom.Node.copyright:Copyright (c) 2023 The Protobom Authors:protobom.protobom.Node.file_name:test.tar.gz:protobom.protobom.Node.hashes:3:b51261db1ecadecf85274e811e537c5811a0ad1ab2a0121aeac4e3d031e1bf83:protobom.protobom.Node.id:node-id:protobom.protobom.Node.license_comments:Comment on the license:protobom.protobom.Node.license_concluded:Apache-2.0:protobom.protobom.Node.name:test node:protobom.protobom.Node.primary_purpose[0]:5:protobom.protobom.Node.release_date:1700080498:protobom.protobom.Node.url_download:http://example.com/file.tar.gz:protobom.protobom.Node.url_home:http://example.com/:protobom.protobom.Node.valid_until_date:1700253298:protobom.protobom.Node.version:1.0.0:supplier:n(ACME, Inc)o(true)email(acme@example.com)url(http://acme-fixtures.com)p(123-3456)",
```
Comment (eddiezane) [inline pkg/sbom/node_test.go:235]: @puerco this test required changing the ordering of the output to pass. I want to think that's expected by the alphabetical sorting but you should check me.

**Comment by puerco** [comment]
Thanks for handling this @eddiezane :pray: 
Once we coordinate the next release in the next few days we'll merge it .

**Comment by jhoward-lm** [comment]
@eddiezane @puerco

Tested locally on my protobom/storage fork branch:

- added go.mod replace to point at this PR branch
- updated all references of `github.com/bom-squad/protobom` to `github.com/protobom/protobom` across the code
- deleted `pkg/options/options.go` and `model/v1/storage/backend.go`
- modified a few of my `Backend` types to reference the new locations of `StoreRetriever` and `ProtobomType`

Looks like it's all working so far!

**Review by puerco — APPROVED** [review]
OK, all tests are passing and local testing works for me too. Let's merge it :rocket: 

Great work @eddiezane!! And thanks @jhoward-lm for helping to reshape this.

**Pull request merged**